lib/cim/cim-adapter.ts

```diff
@@ -0,0 +1,49 @@
+import {PimClass} from "../pim/model";
+import {CoreResourceReader} from "../core";
+import {IriProvider} from "./iri-provider";
+
+export interface CimAdapter {
+
+    /**
+     * Resources returned by the adapter are on the PIM level, therefore the PIM IRIs are needed to interconnect them.
+     * Also, the IRIs need to be predictable and revertible so the user is able to get CIM iri from PIM iri.
+     * @param iriProvider
+     */
+    setIriProvider(iriProvider: IriProvider): void;
+
+    /**
+     * Searches the CIM and returns an ordered list of PimClasses.
+     * @param searchQuery Search query
+     * @return List of PimClasses matching the search
+     */
+    search(searchQuery: string): Promise<PimClass[]>;
+
+    /**
+     * Returns a PimClass by CIM iri.
+     * @param cimIri CIM iri of class to be interpreted.
+     * @return Interpreted PIM class or null if the requested CIM class does not exists
+     */
+    getClass(cimIri: string): Promise<PimClass | null>;
+
+    /**
+     * Get attributes and associations (in distance 1) to a PIM class from CIM specified by CIM iri.
+     * @param cimIri iri of CIM entity representing PIM class for which we look for attributes and associations
+     * @return Store containing the PSM class and its surroundings
+     */
+    getSurroundings(cimIri: string): Promise<CoreResourceReader>;
```

Comment:
I would consider using ```neighbors``` as it has this location specific meaning. Perhaps with ```adjacent```  as an adjective? The point is to get the information about the distance in the name of the method.